include/ola/rdm/RDMCommand.h

```diff
@@ -250,74 +215,109 @@ class RDMCommand {
  */
 class RDMRequest: public RDMCommand {
  public:
-    RDMRequest(const UID &source,
-               const UID &destination,
-               uint8_t transaction_number,
-               uint8_t port_id,
-               uint8_t message_count,
-               uint16_t sub_device,
-               RDMCommandClass command_class,
-               uint16_t param_id,
-               const uint8_t *data,
-               unsigned int length):
-      RDMCommand(source,
-                 destination,
-                 transaction_number,
-                 port_id,
-                 message_count,
-                 sub_device,
-                 param_id,
-                 data,
-                 length),
-      m_command_class(command_class) {
-    }
-
-    RDMCommandClass CommandClass() const { return m_command_class; }
-    uint8_t PortId() const { return m_port_id; }
-
-    virtual RDMRequest *Duplicate() const {
-      return DuplicateWithControllerParams(
-        SourceUID(),
-        TransactionNumber(),
-        PortId());
-    }
-
-    virtual RDMRequest *DuplicateWithControllerParams(
-        const UID &source,
-        uint8_t transaction_number,
-        uint8_t port_id) const {
-      return new RDMRequest(
-        source,
-        DestinationUID(),
-        transaction_number,
-        port_id,
-        MessageCount(),
-        SubDevice(),
-        m_command_class,
-        ParamId(),
-        ParamData(),
-        ParamDataSize());
-    }
-
-    virtual void Print(CommandPrinter *printer,
-                       bool summarize,
-                       bool unpack_param_data) const {
-      printer->Print(this, summarize, unpack_param_data);
-    }
-
-    /**
-     * @brief Check if this is a DUB request.
-     * @returns true if this is a DUB request.
-     */
-    bool IsDUB() const;
-
-    // Convert a block of data to an RDMCommand object
-    static RDMRequest* InflateFromData(const uint8_t *data,
-                                       unsigned int length);
-    static RDMRequest* InflateFromData(const std::string &data);
+  /**
+   * @brief Create a new request.
+   * @param source The source UID.
+   * @param destination The destination UID.
+   * @param transaction_number The transaction number.
+   * @param port_id The Port ID.
+   * @param message_count Set to 0.
+   * @param sub_device The Sub Device index.
+   * @param command_class The Command Class of this request.
+   * @param param_id The PID value.
+   * @param data The parameter data, or NULL if there isn't any.
+   * @param length The length of the parameter data.
+   */
+  RDMRequest(const UID &source,
+             const UID &destination,
+             uint8_t transaction_number,
+             uint8_t port_id,
+             uint8_t message_count,  // TODO(simon): remove since it's 0
```

Comment:
No, I plan on breaking this.

Popping up a level, we need to think about what APIs we commit to (OlaClient, DmxBuffer) and what we're free to change. I don't believe we should be supporting everything in include/ola as a public API.